engine/src/main/java/org/terasology/engine/subsystem/openvr/OpenVRControllers.java

```diff
@@ -51,90 +59,83 @@
-        } else if (OpenVRUtil.switchedDown(ControllerListener.BUTTON_GRIP, stateBefore.ulButtonPressed, stateAfter.ulButtonPressed)) {
-            queuedActions.add(new ControllerAction(InputType.CONTROLLER_BUTTON.getInput(ControllerId.ONE),
-                    "OpenVR", ButtonState.DOWN, 1.0f));
-        } else if (OpenVRUtil.switchedUp(ControllerListener.BUTTON_APP_MENU, stateBefore.ulButtonPressed, stateAfter.ulButtonPressed)) {
-            queuedActions.add(new ControllerAction(InputType.CONTROLLER_BUTTON.getInput(ControllerId.TWO),
-                    "OpenVR", ButtonState.UP, 1.0f));
-        } else if (OpenVRUtil.switchedDown(ControllerListener.BUTTON_APP_MENU, stateBefore.ulButtonPressed, stateAfter.ulButtonPressed)) {
-            queuedActions.add(new ControllerAction(InputType.CONTROLLER_BUTTON.getInput(ControllerId.TWO),
-                    "OpenVR", ButtonState.DOWN, 1.0f));
-        } else if (OpenVRUtil.switchedDown(ControllerListener.BUTTON_TOUCHPAD, stateBefore.ulButtonPressed, stateAfter.ulButtonPressed)) {
-            queuedActions.add(new ControllerAction(InputType.CONTROLLER_AXIS.getInput(ControllerId.X_AXIS),
-                    "OpenVR", ButtonState.DOWN, -stateAfter.rAxis[0].x));
-            queuedActions.add(new ControllerAction(InputType.CONTROLLER_AXIS.getInput(ControllerId.Y_AXIS),
-                    "OpenVR", ButtonState.DOWN, stateAfter.rAxis[0].y));
-        } else if (OpenVRUtil.switchedUp(ControllerListener.BUTTON_TOUCHPAD, stateBefore.ulButtonPressed, stateAfter.ulButtonPressed)) {
-            queuedActions.add(new ControllerAction(InputType.CONTROLLER_AXIS.getInput(ControllerId.X_AXIS),
-                    "OpenVR", ButtonState.UP, 0));
-            queuedActions.add(new ControllerAction(InputType.CONTROLLER_AXIS.getInput(ControllerId.Y_AXIS),
-                    "OpenVR", ButtonState.UP, 0));
+    private boolean switchedUp(long buttonIndex) {
+        return OpenVRUtil.switchedUp(buttonIndex, cachedStateBefore.ulButtonPressed, cachedStateAfter.ulButtonPressed);
+    }
+
+    private boolean switchedDown(long buttonIndex) {
+        return OpenVRUtil.switchedDown(buttonIndex, cachedStateBefore.ulButtonPressed, cachedStateAfter.ulButtonPressed);
+    }
+
+    private void addAxisAction(int controllerButton, ButtonState buttonState, float axisValue) {
+        queuedActions.add(new ControllerAction(InputType.CONTROLLER_AXIS.getInput(controllerButton),
+                    "OpenVR", buttonState, axisValue));
+    }
+
+    private void addButtonAction(int controllerButton, ButtonState buttonState) {
+        queuedActions.add(new ControllerAction(InputType.CONTROLLER_BUTTON.getInput(controllerButton),
+                "OpenVR", buttonState, 1.0f));
+    }
+
+    private void handleController0() {
+        if (switchedUp(ControllerListener.BUTTON_TRIGGER)) {
+            addButtonAction(ControllerId.ZERO, ButtonState.UP);
+        } else if (switchedDown(ControllerListener.BUTTON_TRIGGER)) {
```

Comment:
Well, `ControllerId.ZERO` is part of Terasology's controller interface. `BUTTON_TRIGGER` comes from the following line:

```
long BUTTON_TRIGGER = (1L << JOpenVRLibrary.EVRButtonId.EVRButtonId_k_EButton_SteamVR_Trigger);
```

which I wrote. the programmer-esque word order comes from the original SDK, and I converted it over to our check style guidelines.